utest/meson.build

```diff
@@ -104,7 +104,8 @@ openblas_utest_ext_exec = executable(
     # `common.h` from the source root, not `utest/test_extensions` directory
     # one. Also, `config.h` is needed that lives in the build directory.
     implicit_include_directories: false,
-    c_args: _cargs + ['-I..', '-I../build'],
+    include_directories: _inc,
+    c_args: _cargs + [f'-I@global_bld_dir@'],
```

Comment:
@HaoZeke So the case is that `include_directories: _inc` includes both `-I..` and `-I.`. The `-I.` is undesired as is breaks the build. I think this might be an issue in meson (that both paths are included) but I think we need to rely on `c_args: _cargs + ['-I..', f'-I@global_bld_dir@']` instead. WDYT?

Reply:
I was under the impression that `_inc` should only contain `-I..`, since `utest_inc` is the one with `-I.`

Reply:
I would expect the same but with `_inc` `-I.` is also present:
```
FAILED: utest/openblas_utest_ext.p/test_extensions_common.c.o 
cc -Iutest/openblas_utest_ext.p -I/home/runner/work/OpenBLAS/OpenBLAS/build -I. -I.. 
-fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99
-O3 -DSMALL_MATRIX_OPT -DMAX_STACK_ALLOC=2048 -mmmx -msse -msse2
-msse3 -mssse3 -msse4.1 -msse4.2 -mfma -mavx -mavx2 -pthread -m64
-DSMP_SERVER -DBUILD_SINGLE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1
-DBUILD_COMPLEX16=1 -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME
-UCHAR_CNAME -DNO_AFFINITY -DNO_WARMUP -MD -MQ
utest/openblas_utest_ext.p/test_extensions_common.c.o -MF
utest/openblas_utest_ext.p/test_extensions_common.c.o.d
-o utest/openblas_utest_ext.p/test_extensions_common.c.o
-c ../utest/test_extensions/common.c
```
My guess it's a bug (or intended behavior) in meson.

Reply:
Thinking about this a bit more `global` wasn't right anyway, the latest should pass the build robustly using `project` scoped variables.